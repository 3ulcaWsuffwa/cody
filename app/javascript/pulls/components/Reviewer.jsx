// @flow

import React from "react";
import { createFragmentContainer, graphql } from "react-relay";

function statusToOcticon(status: string) {
  switch (status) {
    case "pending_approval":
      return (
        <span className="icon">
          <i className="fa fa-square-o" />
        </span>
      );
    case "approved":
      return (
        <span className="icon">
          <i className="fa fa-check-square-o" />
        </span>
      );
    default:
      return status;
  }
}

const Reviewer = ({ reviewer }) =>
  <div className="level">
    <div className="level-left">
      <div className="level-item">
        {statusToOcticon(reviewer.status)}
      </div>
      <div className="level-item">
        <strong>
          {reviewer.login}
        </strong>
      </div>
      <div className="level-item">
        {reviewer.reviewRule != null ? reviewer.reviewRule.name : false}
      </div>
    </div>
  </div>;

export default createFragmentContainer(
  Reviewer,
  graphql`
    fragment Reviewer_reviewer on Reviewer {
      id
      login
      status
      reviewRule {
        name
      }
    }
  `
);
