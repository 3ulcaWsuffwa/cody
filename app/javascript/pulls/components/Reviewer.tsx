import React from "react";
import { createFragmentContainer, graphql } from "react-relay";
import type {
  Reviewer_reviewer,
  ReviewerStatus
} from "./__generated__/Reviewer_reviewer.graphql";

function statusToOcticon(status: ReviewerStatus) {
  switch (status) {
    case "PENDING_APPROVAL":
      return (
        <span className="icon color-warning" title="Pending approval">
          <i className="far fa-circle" />
        </span>
      );
    case "APPROVED":
      return (
        <span className="icon color-success" title="Approved">
          <i className="fas fa-circle" />
        </span>
      );
    default:
      return status;
  }
}

const Reviewer = ({ reviewer }: { reviewer: Reviewer_reviewer }) => (
  <div className="level">
    <div className="level-left">
      <div className="level-item">{statusToOcticon(reviewer.status)}</div>
      <div className="level-item">
        <strong>{reviewer.login}</strong>
      </div>
      <div className="level-item">
        {reviewer.reviewRule != null ? reviewer.reviewRule.name : false}
      </div>
    </div>
  </div>
);

export default createFragmentContainer(
  Reviewer,
  {
    reviewer: graphql`
      fragment Reviewer_reviewer on Reviewer {
        id
        login
        status
        reviewRule {
          name
        }
      }
    `
  }
);
