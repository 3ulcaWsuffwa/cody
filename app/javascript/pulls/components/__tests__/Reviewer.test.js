import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Reviewer from "../Reviewer";

test("Reviewer snapshot test", () => {
  const props = {
    login: "aergonaut",
    reviewRule: {
      name: "Foo Review"
    }
  };
  const component = renderer.create(<Reviewer reviewer={props} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Reviewer renders the reviewer's login", () => {
  const component = shallow(<Reviewer reviewer={{ login: "aergonaut" }} />);
  expect(component.text()).toMatch("aergonaut");
});

test("Reviewer renders the review rule name if it is given", () => {
  const props = {
    login: "aergonaut",
    reviewRule: {
      name: "Foo Review"
    }
  };
  const component = shallow(<Reviewer reviewer={props} />);
  expect(component.text()).toMatch(props.reviewRule.name);
});
