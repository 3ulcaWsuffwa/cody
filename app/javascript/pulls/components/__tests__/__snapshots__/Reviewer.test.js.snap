// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviewer snapshot test 1`] = `
<div
  className="level"
>
  <div
    className="level-left"
  >
    <div
      className="level-item"
    />
    <div
      className="level-item"
    >
      <strong>
        aergonaut
      </strong>
    </div>
    <div
      className="level-item"
    >
      Foo Review
    </div>
  </div>
</div>
`;
